src/ops/constant.rs

```diff
@@ -227,7 +227,8 @@ pub trait CustomConst:
     fn name(&self) -> SmolStr;
 
     /// Returns the type of the constant.
-    fn const_type(&self) -> ClassicType;
+    // TODO it would be good to ensure that this is a *classic* CustomType not a linear one!
+    fn const_type(&self) -> CustomType;
```

Comment:
Note this is a different method from the `pub fn const_type` (not part of any trait) on line 118 of this file

Reply:
Should we rename one or the other??

Reply:
I think the `ConstValue` one could just be renamed to `type`, since we already know it's const

Reply:
This one could be `custom_type`

Reply:
I tried `type`....but of course that's a reserved word! So `custom_type` seems good, thanks for the suggestion :)